homeassistant/components/switch/rpi_servo.py

```diff
@@ -0,0 +1,127 @@
+"""
+Allows to configure a switch using RPi GPIO Servos.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.rpi_gpio/
+"""
+import logging
+
+import voluptuous as vol
+import homeassistant.components.rpi_gpio as gpio
+from homeassistant.helpers.event import track_point_in_time
+from homeassistant.components.switch import PLATFORM_SCHEMA
+from homeassistant.const import DEVICE_DEFAULT_NAME
+from homeassistant.helpers.entity import ToggleEntity
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['rpi_gpio']
+
+CONF_PORTS = 'ports'
+CONF_INACTIVE_POSTION = "default_position"
+CONF_POSITION_ON = "position_on"
+CONF_POSITION_OFF = "position_off"
+CONF_POSITION_DURATION = "enabled_duration"
+DEFAULT_INACTIVE_POSITION = 90
+DEFAULT_POSITION_ON = 180
+DEFAULT_POSITION_OFF = 0
+DEFAULT_POSITION_DURATION = 1
+
+_SWITCHES_SCHEMA = vol.Schema({
+    cv.positive_int: cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_PORTS): _SWITCHES_SCHEMA,
+    vol.Optional(CONF_INACTIVE_POSTION,
+                 default=DEFAULT_INACTIVE_POSITION): cv.positive_int,
+    vol.Optional(CONF_POSITION_ON,
+                 default=DEFAULT_POSITION_ON): cv.positive_int,
+    vol.Optional(CONF_POSITION_OFF,
+                 default=DEFAULT_POSITION_OFF): cv.positive_int,
+    vol.Optional(CONF_POSITION_DURATION,
+                 default=DEFAULT_POSITION_DURATION): cv.positive_int,
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Raspberry PI GPIO servo devices."""
+    inactive_position = config.get(CONF_INACTIVE_POSTION)
+    position_on = config.get(CONF_POSITION_ON)
+    position_off = config.get(CONF_POSITION_OFF)
+    enabled_duration = config.get(CONF_POSITION_DURATION)
+    switches = []
+    ports = config.get(CONF_PORTS)
+    for port, name in ports.items():
+        switches.append(RPiGPIOServo(hass, name, port, inactive_position,
+                                     position_on, position_off,
+                                     enabled_duration))
+    add_devices(switches)
+
+
+class RPiGPIOServo(ToggleEntity):
+    """Representation of a  Raspberry Pi GPIO Servo."""
+
+    def __init__(self, hass, name, port, inactive_position, position_on,
```

Comment:
Don't pass in `hass`, it will be set on the entity when it has been added to home assistant. After that time, you can access `hass` with `self.hass`.